frontend/hooks/useExpoToken.js

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react';
+import * as Device from 'expo-device';
+import * as Notifications from 'expo-notifications';
+
+async function registerForPushNotificationsAsync() {
+    let token;
+    if (Device.isDevice) {
+        const { status: existingStatus } = await Notifications.getPermissionsAsync();
+        let finalStatus = existingStatus;
+        if (existingStatus !== 'granted') {
+            const { status } = await Notifications.requestPermissionsAsync();
+            finalStatus = status;
+        }
+        if (finalStatus !== 'granted') {
+            alert('Failed to get push token for push notification!');
+            return;
+        }
+        token = await Notifications.getExpoPushTokenAsync();
+        console.log(token);
+    } else {
+        alert('Must use physical device for Push Notifications');
+    }
+
+    if (Platform.OS === 'android') {
+        Notifications.setNotificationChannelAsync('default', {
+            name: 'default',
+            importance: Notifications.AndroidImportance.MAX,
+            vibrationPattern: [0, 250, 250, 250],
+            lightColor: '#FF231F7C',
+        });
+    }
+
+    return token;
+}
+export default function useExpoToken() {
```

Comment:
actually we don't use this custom hook anywhere
but we have it here because it is tightly coupled to useNotificationNote hook
useExpoToken hook asks user for permissions to send notifications if we don't have them already
then this hook returns the token to uniquely identify an app instance to send notification to
we can store it on backend side in database to sent specific notifications to specific users